honeybee_vtk/writer.py

```diff
@@ -19,23 +19,59 @@ def writer(file_path, *, file_name=None, target_folder=None, include_grids=False
             file name for the .zip file.
         target_folder: A text string to a folder to write the output file. The file
             will be written to the current folder if not provided.
-        include_grids: A boolean. Defaults to False. Grids will not be extracted from
-            HBJSON if set to False.
-        include_vectors: A boolean. Defaults to False. Vector arrows will not be created
-            if set to False.
-        include_points: A boolean. Defaults to False. If set to True, sensor points
-            color-grouped based on their vector direction will be exported instead of
-            arrows for the vectors.
         writer: A text string to indicate the VTK writer. Acceptable values are
             'vtk', 'xml', and 'html'. Defaults to 'html'.
         open_html: A boolean. If set to True, it will open the generated HTML
             in a web browser when 'html' is provided as value in the writer argument.
             Defaults to False.
-
+        include_grids: A boolean. Grids will be exported if the value is True. Here, a
+            grid is exported as base-geometry, grid mesh, or sensorpoints in that order
+            of preference. So if a Sensorgrid object in HBJSON has base-geometry, then 
+            that object will be exported as base geometry, eventhough it also might have
+            mesh and sensors.
+        include_sensors: A text string to indicate whether to show sensor directions as
+            vectors or points colored based on directions. Acceptable values are;
+            'vectors' and 'points.' Defaults to False.
+        include_normals: A text string to indicate whether to show face normals as
+            vectors or points colored based on directions. Acceptable values are;
+            'vectors' and 'points.' Defaults to False. Currently, this function
+            only exports normals for the apertures. Other face types will be supported
+            in future.
+            
     Returns:
-        A text string containing the path to the output file.
+        A tuple of two elements.
+
+        - hbjson: A dictionary.
+
+        - writer_val: A text string.
     """
 
+    # Validate file_name
+    if file_name:
+        if not isinstance(file_name, str):
+            raise ValueError('file_name needs to be a text string.')
+
+    # Validate open_html
+    if not isinstance(open_html, bool):
+        raise ValueError('Open_html only accepts a boolean value.')
+
+    if not isinstance(include_grids, bool):
+        raise ValueError('include_grids only accepts a boolean value.')
+
+    # Validate include_sensors
+    if include_sensors:
+        if not isinstance(include_sensors, str) or include_sensors.lower()\
+            not in ['vectors', 'points']:
+            raise ValueError('The value for include_sensors can be either "vectors"'
+                             ' or "points".')
```

Comment:
Removing False as an input and using None will make your life easier. I really don't get all these type checks for string inputs. To make sure it's vectors or points makes sense. Just check if the item is inside the list and if not raise an error.